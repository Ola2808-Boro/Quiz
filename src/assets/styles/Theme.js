export const theme = {
  colors: {
    backgroundColor: "#F5F7FB",
    text: "#293264",
    buttonColor: "#F5F7FB",
    circleBottom: "#DEEBF8",
    circleUp: "#FFFAD1",
    buttonBg: "#4D5B9E",
    borderAnswer: "#4D5B9E",
  },

  fontSize: {
    huge: "70px",
    xlgrand: "44px",
    lgrand: "40px",
    grand: "37px",
    xxxxl: "34px",
    xxxl: "33px",
    xxl: "30 px",
    xl: "26px",
    mediumxl: "24px",
    l: "22px",
    ml: "18px",
    m: "14px",
    s: "10px",
  },
};
