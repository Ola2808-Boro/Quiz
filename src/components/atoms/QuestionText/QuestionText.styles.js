import styled from "styled-components";
import { device } from "helpers/breakpoints";
export const StyledQuestion = styled.p`
  color: ${({ theme }) => theme.colors.text};
  font-size: ${({ theme }) => theme.fontSize.ml};
  padding-bottom: 0.7rem;

  @media ${device.mobileS} {
    font-size: ${({ theme }) => theme.fontSize.l};
    padding-bottom: 0.5rem;
    word-break: break-word;
  }
  @media ${device.tablet} {
    font-size: ${({ theme }) => theme.fontSize.xl};
  }
  @media ${device.laptop} {
    font-size: ${({ theme }) => theme.fontSize.xxl};
  }
  @media ${device.desktop} {
    font-size: ${({ theme }) => theme.fontSize.xxxl};
  }
`;
