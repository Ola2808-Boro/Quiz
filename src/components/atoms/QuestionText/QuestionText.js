import React from "react";
import PropTypes from "prop-types";
import { StyledQuestion } from "./QuestionText.styles";

const QuestionText = ({ question, number }) => {
  return (
    <StyledQuestion>
      {number} {question}
    </StyledQuestion>
  );
};

QuestionText.propTypes = {
  question: PropTypes.string.isRequired,
  number: PropTypes.string.isRequired,
};

export default QuestionText;
