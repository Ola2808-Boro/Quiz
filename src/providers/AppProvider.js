import React from 'react';
import { GlobalStyle } from 'assets/styles/GlobalsStyle'
import { ThemeProvider } from 'styled-components';
import 'assets/styles/fonts.css'
import {theme} from 'assets/styles/Theme'


const AppProviders = ({ children }) => {
    return (
 
          <ThemeProvider theme={theme}>      
                <GlobalStyle />
                {children}
          </ThemeProvider>
 
    );
  };
  
  export default AppProviders;