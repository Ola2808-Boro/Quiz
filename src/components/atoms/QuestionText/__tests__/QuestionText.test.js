import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import AppProvider from "providers/AppProvider";
import { render, screen } from "@testing-library/react";
import QuestionText from "../QuestionText";
import { StyledQuestion } from "components/atoms/QuestionText/QuestionText.styles";
describe("Test style", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppProvider>
          <StyledQuestion />
        </AppProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("test question content", () => {
    render(
      <AppProvider>
        <QuestionText question="Question" number="1" />
      </AppProvider>
    );
    const questionElement = screen.getByText(/question/i);
    expect(questionElement).toBeInTheDocument();
  });
});
