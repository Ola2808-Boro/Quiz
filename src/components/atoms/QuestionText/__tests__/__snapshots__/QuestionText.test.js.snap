// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test style default style 1`] = `
<p
  className="ddyCMY"
/>
`;
